### PR TITLE
feat(hub-common): add updated utils for managing and previewing feeds

### PR DESCRIPTION
**Pull request by sonofflynn89**
affects: @esri/hub-common

1. Description:
Part of https://devtopia.esri.com/dc/hub/issues/11611

Adds new helpers for managing feed configurations, such as
- getFeedTemplate()
- setFeedTemplate()
- previewFeedTemplate()

We want to use these new functions instead of the older `getFeedConfiguration()` and `setFeedConfiguration()` as we are moving more and more towards passing around HubEntity objects instead of IModel objects.

Also note, `getFeedTemplate()` will return the correct default template if none is configured. This was a change from `getFeedConfiguration()`

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

### CODE REVIEW

File: packages/common/src/sites/feeds/previewFeed.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  const baseUrl = `${context.hubUrl}/api/feed/${format}/${version}`;
+
+  const searchParams = new URLSearchParams();
+  searchParams.set("id", previewHubId);
+
+  // The feeds api has different query parameters for rss and dcat feeds
+  format === "rss"
+    ? searchParams.set("rssConfig", stringifiedTemplate)
+    : searchParams.set("dcatConfig", stringifiedTemplate);
```
Comment (abp6318) [inline packages/common/src/sites/feeds/previewFeed.ts:33]:
```suggestion
  searchParams.set(format === "rss" ? "rssConfig" : "dcatConfig", stringifiedTemplate);
```

File: packages/common/src/sites/feeds/previewFeed.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * @param opts.previewTemplate - the feed template to preview
+ * @param opts.previewHubId - the Hub ID to preview the feed for
+ * @param opts.context - the ArcGIS context
+ * @returns the previewed feed as a string
+ */
+export async function previewFeed(opts: IPreviewFeedOptions): Promise<string> {
+  const { format, version, previewHubId, context } = opts;
+  const stringifiedTemplate = JSON.stringify(opts.previewTemplate);
+
+  const baseUrl = `${context.hubUrl}/api/feed/${format}/${version}`;
```
Comment (abp6318) [inline packages/common/src/sites/feeds/previewFeed.ts:25]: I've made a mistake in the past using a URL that could occasionally include a different base route for the API or include query parameters. That can't happen here, can it?

File: packages/common/src/sites/feeds/previewFeed.ts (hunk 1 above)
Comment (abp6318) [reply packages/common/src/sites/feeds/previewFeed.ts:33]: Better yet, you use `format === "rss"` as well below, you could just make it a constant

File: packages/common/test/sites/feeds/_internal/getMajorVersion.test.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    const version = "";
+    const result = getMajorVersion(version);
+    expect(result).toBe("");
+  });
+
+  it("should return the major version for a version with non-numeric characters", () => {
+    const version = "v7.8.9";
+    const result = getMajorVersion(version);
+    expect(result).toBe("v7");
+  });
```
Comment (abp6318) [inline packages/common/test/sites/feeds/_internal/getMajorVersion.test.ts:50]: Oh slay, glad you thought of this test

**Review by abp6318 — COMMENTED** [review]
Looks good! Hestitant to approve until I've had a chance to test this out with the frontend.

File: packages/common/src/sites/feeds/previewFeed.ts (hunk 2 above)
Comment (abp6318) [reply packages/common/src/sites/feeds/previewFeed.ts:25]: Searching hub.js I see `getHubApiUrl`, `getHubApiUrlFromPortal`, and `getHubApiUrlRoot`

**Comment by abp6318** [comment]
For other reviewers, you can test on my site here: https://qa-pre-a-hub.hubqa.arcgis.com/workspace/sites/8e67c41951804ef9965e290b3d83c182/content?uiVersion=44644e904f6ac59d41098fae6b896a50a142351e&pe=temp:hub:site:workspace:feeds

Need to be logged in as `paige_pa` or `qa_pre_a_hub_admin`

File: packages/common/test/sites/feeds/_internal/getMajorVersion.test.ts (hunk 3 above)
Comment (sonofflynn89) [reply packages/common/test/sites/feeds/_internal/getMajorVersion.test.ts:50]: Copilot was the one that actually thought of it 😂 But I'm glad it thought of it too!

File: packages/common/src/sites/feeds/previewFeed.ts (hunk 2 above)
Comment (sonofflynn89) [reply packages/common/src/sites/feeds/previewFeed.ts:25]: @abp6318 excellent question! I talked with @tomwayson and we are good to use `context.hubUrl` in this scenario. All the functions you mentioned above pre-date `IArcGISContext`, so we don't want to rely on them if we don't have to.

**Review by abp6318 — APPROVED** [review]
LGTM -- send it 🛹

**Pull request merged**